chain-crypto/src/algorithms/vrf/vrf.rs

```diff
@@ -100,34 +100,50 @@ impl SecretKey {
 
     /// Create a proof, for the given parameters; no check is made to make sure it's correct
     ///
-    /// the proof is randomized, so need a secure RNG.
+    /// the proof is randomized, so need a freshly randomly scalar for random.
+    /// use 'prove_simple' to use a RNG and avoid generating this random.
     ///
-    /// use 'evaluate' for creating the proof directly from input
-    fn prove<T: RngCore + CryptoRng>(
+    /// use 'evaluate' or 'evaluate_simple' for creating the proof directly from input
+    pub fn prove(&self, r: &Scalar, m_point: GroupElement, output: OutputSeed) -> ProvenOutputSeed {
+        let dleq = dleq::Dleq {
+            g1: &GroupElement::generator(),
+            h1: &self.public,
+            g2: &m_point,
+            h2: &output.0,
+        };
+        let dleq_proof = dleq::generate(&r, &self.secret, &dleq);
+        ProvenOutputSeed {
+            u: output.clone(),
+            dleq_proof,
+        }
+    }
+
+    pub fn prove_simple<T: RngCore + CryptoRng>(
```

Comment:
I would use the latest version of this as well. Remove the possibility of inputing the randomness at call time. This should never be required.